pkg/authenticator/authenticator.go

```diff
@@ -112,28 +112,65 @@ func (auth *Authenticator) Login() error {
 
 	log.InfoLogger.Printf(log.CAKC007I, auth.Config.Username)
 
-	csrRawBytes, err := auth.GenerateCSR(auth.Config.Username.Suffix)
+	// The cert file may be present from a previous login request
+	if !auth.CertFileExist() {
+		csrRawBytes, err := auth.GenerateCSR(auth.Config.Username.Suffix)
 
-	csrBytes := pem.EncodeToMemory(&pem.Block{
-		Type: "CERTIFICATE REQUEST", Bytes: csrRawBytes,
-	})
+		csrBytes := pem.EncodeToMemory(&pem.Block{
+			Type: "CERTIFICATE REQUEST", Bytes: csrRawBytes,
+		})
 
-	req, err := LoginRequest(auth.Config.URL, auth.Config.ConjurVersion, csrBytes, auth.Config.Username.Prefix)
-	if err != nil {
-		return err
+		req, err := LoginRequest(auth.Config.URL, auth.Config.ConjurVersion, csrBytes, auth.Config.Username.Prefix)
+		if err != nil {
+			return err
+		}
+
+		resp, err := auth.client.Do(req)
+		if err != nil {
+			return log.RecordedError(log.CAKC028E, err.Error())
+		}
+
+		err = EmptyResponse(resp)
+		if err != nil {
+			return log.RecordedError(log.CAKC029E, err.Error())
+		}
 	}
 
-	resp, err := auth.client.Do(req)
+	err := auth.LoadClientCert()
 	if err != nil {
-		return log.RecordedError(log.CAKC028E, err.Error())
+		return err
 	}
 
-	err = EmptyResponse(resp)
-	if err != nil {
-		return log.RecordedError(log.CAKC029E, err.Error())
+	return nil
+}
+
+// Returns true if we are logged in (have a cert)
+func (auth *Authenticator) IsLoggedIn() bool {
+	return auth.PublicCert != nil
+}
+
+// CertFileExist returns true if certificate file exists in the ClientCertPath
+func (auth *Authenticator) CertFileExist() bool {
+	info, err := os.Stat(auth.Config.ClientCertPath)
+	if os.IsNotExist(err) {
+		return false
 	}
+	return !info.IsDir()
```

Comment:
this return seems a bit weird to me because in the above if statement, you are returning false if it was reported that a file or directory does not exist but then you are returning the certPath is not a file/directory with `!info.IsDir()` which is unreachable.

Shouldn't you return a condition that evaluates to true?